### PR TITLE
Plant icon unit tests

### PR DESCRIPTION
**Pull request by silicons**
removes the growth icon test from adv proccall as there's basically no reason to use it in game if it's already done in travis when people pr. even though i normally hate removing admeme procs.

### CODE REVIEW

**Review by Cyberboss — DISMISSED** [review]

**Comment by Cyberboss** [comment]
So it turns out it was broken after all

**Comment by ShizCalev** [comment]
lmao

File: code/modules/unit_tests/plant_growth_icons.dm [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/datum/unit_test/plant_growth_icons
+
+/datum/unit_test/plant_growth_icons/Run()
+	var/list/states = icon_states('icons/obj/hydroponics/growing.dmi')
+	states |= icon_states('icons/obj/hydroponics/growing_fruits.dmi')
+	states |= icon_states('icons/obj/hydroponics/growing_flowers.dmi')
+	states |= icon_states('icons/obj/hydroponics/growing_mushrooms.dmi')
+	states |= icon_states('icons/obj/hydroponics/growing_vegetables.dmi')
```
Comment (ShizCalev) [inline code/modules/unit_tests/plant_growth_icons.dm]: add goon\icons\obj\hydroponics.dmi

**Comment by ShizCalev** [comment]
/obj/item/seeds/lavaland is a dummy parent, just assign it a random grow_icon and you should be fine

**Comment by kevinz000** [comment]
changed it to respect the seed growing icon files, as opposed to searching all of the states.

**Comment by Cyberboss** [comment]
```
REASON #600 ...
```

You made it worse

File: code/modules/hydroponics/grown/mushrooms.dm [hunk 2]
```diff
@@ -322,6 +322,7 @@
 	rarity = 20
 	reagents_add = list("nutriment" = 0.1)
 	resistance_flags = FIRE_PROOF
+	growing_icon = 'icons/obj/hydroponics/growing_mushrooms.dmi'
```
Comment (ShizCalev) [inline code/modules/hydroponics/grown/mushrooms.dm]: species

Comment (kevinz000) [reply code/modules/hydroponics/grown/mushrooms.dm]: oof

Comment (kevinz000) [reply code/modules/hydroponics/grown/mushrooms.dm]: fixed

File: code/modules/hydroponics/grown/mushrooms.dm [hunk 3]
```diff
@@ -366,5 +363,3 @@
 	species = "ember"
 	plantname = "Embershroom Mushrooms"
 	product = /obj/item/reagent_containers/food/snacks/grown/ash_flora/mushroom_stem
-	genes = list(/datum/plant_gene/trait/plant_type/fungal_metabolism, /datum/plant_gene/trait/glow)
```
Comment (ShizCalev) [inline code/modules/hydroponics/grown/mushrooms.dm]: missing the glow trait

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/unit_tests/plant_growth_icons.dm [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+		'icons/obj/hydroponics/growing_fruits.dmi',
+		'icons/obj/hydroponics/growing_flowers.dmi',
+		'icons/obj/hydroponics/growing_mushrooms.dmi',
+		'icons/obj/hydroponics/growing_vegetables.dmi',
+		'goon/icons/obj/hydroponics.dmi'
+		)
+	var/list/states = list()
+	for(var/file in files)
+		states["[file]"] = icon_states(file)
+	var/list/paths = typesof(/obj/item/seeds) - /obj/item/seeds - typesof(/obj/item/seeds/sample)
```
Comment (ninjanomnom) [inline code/modules/unit_tests/plant_growth_icons.dm]: `subtypesof(/obj/item/seeds) - typesof(/obj/item/seeds/sample)`

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Pull request closed without merging**